frontend/src/components/Inquiries/InquiriesTable.tsx

```diff
@@ -1,86 +1,102 @@
-import {useQuery} from "@tanstack/react-query";
-import {SkeletonText, Table, TableContainer, Tbody, Td, Th, Thead, Tr} from "@chakra-ui/react";
-import * as InquiriesService from "../../client/services/inquiriesService.ts";
-import dayjs from "dayjs";
-import utc from "dayjs/plugin/utc";
-import timezone from "dayjs/plugin/timezone";
+import {
+  SkeletonText,
+  Table,
+  TableContainer,
+  Tbody,
+  Td,
+  Th,
+  Thead,
+  Tr,
+} from "@chakra-ui/react"
+import { useQuery } from "@tanstack/react-query"
+import dayjs from "dayjs"
+import timezone from "dayjs/plugin/timezone"
+import utc from "dayjs/plugin/utc"
+import * as InquiriesService from "../../client/services/inquiriesService.ts"
 
 // Dayjs Configurations
-// eslint-disable-next-line
-dayjs.extend(utc);
-// eslint-disable-next-line
-dayjs.extend(timezone);
-// eslint-disable-next-line
-const userTimezone = dayjs.tz.guess();
+/* eslint-disable */
+dayjs.extend(utc)
+dayjs.extend(timezone)
+/* eslint-enable */
 
-function getInquiriesQueryOptions() {
-    return {
-        queryKey: ["inquiries"],
-        queryFn: () => InquiriesService.readInquiries(),
-    };
-}
+const InquiriesTable = () => {
+  // Format ISO date to the user's timezone.
+  // ex. Sep 17, 2024 14:13 PM
+  function formatDate(date: string): string {
+    const invalidDateMessage = "Invalid Date"
+    if (typeof date !== "string") return invalidDateMessage
+    // Unsafe access to 'error' typed value is handled by dayjs' isValid function
+    /* eslint-disable */
+    const parsedDate = dayjs.utc(date)
+    const userTimezone = dayjs.tz.guess()
+    return parsedDate.isValid()
+      ? parsedDate.tz(userTimezone).format("MMM DD, YYYY hh:mm A")
+      : invalidDateMessage
+    /* eslint-enable */
+  }
 
-// Format date to user's timezone
-// ex. Sep 17, 2024 14:13 PM
-function formatDate(date: Date): string {
-    try {
-        // eslint-disable-next-line
-        return dayjs.utc(date).tz(userTimezone).format("MMM DD, YYYY hh:mm A");
-    } catch (error) {
-        console.error("Error formatting date:", error);
-        return new Date(date).toISOString();
+  function getInquiriesQueryOptions() {
+    return {
+      queryKey: ["inquiries"],
+      queryFn: () => InquiriesService.readInquiries(),
     }
-}
+  }
 
-function InquiriesTable() {
-    const { data: inquiries, isPending } = useQuery({
-        ...getInquiriesQueryOptions(),
-    });
+  const { data: inquiries, isPending } = useQuery({
+    ...getInquiriesQueryOptions(),
+  })
 
-    // Sort inquries from Newest to oldest
-    const sortedInquiries = inquiries?.data.sort((a, b) => {
-        return new Date(b.created_at).getTime() - new Date(a.created_at).getTime();
-    });
+  // Sort inquries from Newest to oldest
+  const sortedInquiries = inquiries?.data.sort((a, b) => {
```

Comment:
Consider wrapping this in `useMemo` so that this doesn't run on every render. I think it would be something like
```ts
const sortedInquiries = useMemo(() => {
/* your code here*/
}, [inquiries])

```